subsys/bluetooth/host/gatt.c

```diff
@@ -2856,7 +2856,10 @@ static int bt_gatt_store_cf(struct bt_conn *conn)
 
 	cfg = find_cf_cfg(conn);
 	if (!cfg) {
-		/* No cfg found just cleare it */
+		/* No cfg found, just clear it */
+		BT_DBG("No config for CF");
+		bt_settings_encode_key(key, sizeof(key), "cf",
+				       &conn->le.dst, NULL);
```

Comment:
Id mive the existing code for creates the key before this check that way it actually uses the same logic.